### PR TITLE
[Code] Allow setting an output divison & section type on `CustomFilter` and `CustomWriterFactory`

### PR DESCRIPTION
**Pull request by alexander255**
I use this to do custom builds of BrowsCap JSON cache data that are shipped as part of a browser extension developed by me (size matters in this). If wanted I can also add another patch that implements the other filters as special configurations of the custom filter (reducing code concept duplication).

There is also another patch waiting to be made for allowing setting custom cache item prefix length, but this is the more important one.

### CODE REVIEW

**Comment by mimmi20** [comment]
Please change your PR:

* Don't change the Factory and the Filter. You should compose your writer by your own, if you really only write Json files (or add a new factory).
* Add a FilterCollection which implements the FilterInterface where you may add multiple Filters. Please implement it similar to the WriterCollection.
* Add tests for all things you change.

**Comment by alexander255** [comment]
Thanks for your feedback!

> Don't change the Factory and the Filter. You should compose your writer by your own, if you really only write Json files (or add a new factory).

I'm actually writing PHP INI files then convert them to a JSON *cache* using a patched version of your `browser-json-generator` tool.

But maybe I should elaborte some more on what I'm actually trying to do:
Currently the `browscap-json-generator` tool generates a cache containing *all* the properties known to Browscap… which is a lot of data and too much for my use-case. I therefor have several quick-and-dirty patches in different parts of the codebase in place that reduce that extra size to acceptable levels. These I'm now trying to replace with something upstreamable / reusable for others. My idea for this was to extend the `browscap-json-generator` tool with some command-line flags that allow setting which devisions & sections (`full`, `standard`, `lite`) to add to the output, as well specifing a list of properties that shouldn't be emitted. Is that something that – in pricipal – sounds mergable to you? If yes, what would be the best approach (from your point-of-view) to allow for something like this?

> Add a FilterCollection which implements the FilterInterface where you may add multiple Filters. Please implement it similar to the WriterCollection.

Will do. Thanks for your help!

File: src/Browscap/Filter/CustomFilter.php [hunk 1]
```diff
@@ -16,6 +16,11 @@ class CustomFilter implements FilterInterface
      */
     private $fields = [];
 
+    /**
+     * @var string
```
Comment (mimmi20) [inline src/Browscap/Filter/CustomFilter.php]: Please revert all changes in this filter.

Add a FilterCollection which implements the FilterInterface where you may add multiple Filters. Please implement it similar to the WriterCollection.

File: src/Browscap/Writer/Factory/CustomWriterFactory.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 use Browscap\Data\PropertyHolder;
 use Browscap\Filter\CustomFilter;
+use Browscap\Filter\FilterInterface;
```
Comment (mimmi20) [inline src/Browscap/Writer/Factory/CustomWriterFactory.php]: Please revert all change in this factory. Add a new one if nessesary.

**Comment by asgrim** [comment]
Closing this due to inactivity. Please do re-open if you'd like to work on this again! Thank you.

**Pull request closed without merging**